### PR TITLE
mrc-2981: Add script to install packages and to start R session for pulling dependencies

### PR DESCRIPTION
**Pull request by r-ash**
This PR will
* Add `orderly_install` script which runs R code on orderly and orderly_worker containers used to install R packages
* Add `orderly_r` script which starts an R session in orderly container to use for running `orderly_pull_archive` or `orderly_pull_dependencies`

### CODE REVIEW

File: orderly_install [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+CYAN='\033[0;36m'
+YELLOW='\033[1;33m'
+NC='\033[0m'
+echo "
+"
+echo "Installation complete"
+echo -e "${CYAN}ʕಠᴥಠʔ ${YELLOW}ಠ_ಠ ${GREEN}ᶘಠᴥಠᶅ ${RED}VERY IMPORTANT ${GREEN}ʕಠᴥಠʔ ${YELLOW}ಠ_ಠ ${CYAN}ᶘಠᴥಠᶅ${NC}"
+echo "Update Dockerfile so the package is available after a redeploy"
```
Comment (richfitz) [inline orderly_install]: Provide a link here? `https://github.com/mrc-ide/$INSTANCE-orderly-image`

File: orderly_r [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+instance_name - fertility, naomi or threemc"
+    exit 0
+fi
+
+CONTAINER=`docker ps --filter name=^$1_orderly$ --filter status=running -aq`
+
+if [[ -z $CONTAINER ]]; then
+  echo "Could not find container with name $1_orderly check app is running"
+  exit 0
```
Comment (richfitz) [inline orderly_r]:
```suggestion
  exit 1
```

File: orderly_r [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+set -e
+if [[ $# -eq 0 ]]; then
+    echo "Start an R session in docker container to use orderly_pull_dependencies or orderly_pull_archive
+
+Usage: ./orderly_r instance_name
+
+instance_name - fertility, naomi or threemc"
+    exit 0
```
Comment (richfitz) [inline orderly_r]:
```suggestion
    exit 1
```

File: orderly_install [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -e
+if [[ $# -eq 0 ]]; then
+    echo "Usage: ./orderly_install instance_name 'command'
+    instance_name - fertility or naomi
+    command - The remotes::install_github or install_packages command, remember the quotes"
+    exit 0
```
Comment (richfitz) [inline orderly_install]:
```suggestion
    exit 1
```

File: orderly_install [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+fi
+
```
Comment (richfitz) [inline orderly_install:9]: Recommend doing :
```
INSTANCE=$1
```

and not using `$1` after this point

File: orderly_install [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+CONTAINERS=`docker ps --filter name=$1_orderly* --filter status=running -aq`
+CONTAINER_NAMES=`docker ps --filter name=$1_orderly* --filter status=running --format '{{.Names}}' -a | paste -sd '~ ' | sed 's/~/, /'`
+echo "Running Rscript in containers ${CONTAINER_NAMES}"
+
+for CONTAINER in $CONTAINERS
+do
+    docker exec $CONTAINER Rscript -e $2
```
Comment (richfitz) [inline orderly_install]: If you do `shift` after getting `$1` then you can pass everything though here as `$*` or `$@`

File: orderly_r [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+fi
+
+CONTAINER=`docker ps --filter name=^$1_orderly$ --filter status=running -aq`
+
+if [[ -z $CONTAINER ]]; then
+  echo "Could not find container with name $1_orderly check app is running"
+  exit 0
+fi
+
+docker exec -it $CONTAINER R
```
Comment (richfitz) [inline orderly_r:20]: again, recommend saving `$1` as `INSTANCE`, doing a shift, then forwarding all args here

**Review by richfitz — CHANGES_REQUESTED** [review]

**Comment by r-ash** [comment]
Thanks for feedback, I think I've updated all those things

**Review by richfitz — APPROVED** [review]

**Pull request merged**